mlflow/models/utils.py

```diff
@@ -1247,26 +1315,29 @@ def _enforce_pyspark_dataframe_schema(
     return new_pf_input.drop(*columns_to_drop)
 
 
-def _enforce_datatype(data: Any, dtype: DataType, required=True):
+def _enforce_datatype(data: Any, dtype: DataType, required=True, enforce_param=False):
```

Comment:
`enforce_param` flag looks dangerous; the flag needs to align with whether we pass data or param, which is error prone (no way to stop doing `_enforce_datatype(params, dtype)`).